### ISSUE TITLE
Why do I need to provide secret key to client side code !!

### ISSUE THREAD

**Issue by rahat1994** [Mixpeal/flutter_paypal#3]

**Comment by rahat1994** [comment]
`appBar: AppBar(
          title: Text(widget.title),
        ),
        body: Center(
          child: TextButton(
              onPressed: () => {
                    Navigator.of(context).push(
                      MaterialPageRoute(
                        builder: (BuildContext context) => UsePaypal(
                            sandboxMode: true,
                            clientId:
                                "AW1TdvpSGbIM5iP4HJNI5TyTmwpY9Gv9dYw8_8yW5lYIbCqf326vrkrp0ce9TAqjEGMHiV3OqJM_aRT0",
                            secretKey:
                                "EHHtTDjnmTZATYBPiGzZC_AZUfMpMAzj2VZUeqlFUrRJA_C0pQNCxDccB5qoRQSEdcOnnKQhycuOWdP9",
                            returnURL: "https://samplesite.com/return",
                            cancelURL: "https://samplesite.com/cancel",
                            transactions: const [
                              {
                                "amount": {
                                  "total": '10.12',
                                  "currency": "USD",
                                  "details": {
                                    "subtotal": '10.12',
                                    "shipping": '0',
                                    "shipping_discount": 0
                                  }`

**Comment by MDSADABWASIM** [comment]
That's the requirement of PayPal not this library @rahat1994

**Comment by ibariens** [comment]
I would definetly not recommend to use this package in production, it's a huge security risk.
The client_secret should be only available to your server, where also the checkout links should be generated. 

One could decompile the package, sniff traffic, do a memory dump, whatever method and find the client_id and secret, giving you full access to the sellers paypal account.

**Comment by Mixpeal** [comment]
Haven't really had time to see it like that, do you have a PR fix for this? I will gladly merge it.

**Comment by Mixpeal** [comment]
That's true @MDSADABWASIM , but credentials are required to generate a transaction.

**Comment by ibariens** [comment]
Hi @Mixpeal  appreciate it, but unfortunately I don't see an easy fix.

**Comment by mo-aro-etailer** [comment]
MixPeal try contributing to https://pub.dev/packages/flutter_paypal_native

**Comment by rahat1994** [comment]
ended up using web UI...
